packages/kbn-apm-synthtrace/src/scenarios/service_summary_field_version_dependent.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { ApmFields, apm } from '@kbn/apm-synthtrace-client';
+import { random } from 'lodash';
+import { pipeline, Readable } from 'stream';
+import semver from 'semver';
+import { Scenario } from '../cli/scenario';
+import { deleteSummaryFieldTransform } from '../lib/utils/transform_helpers';
+import { withClient } from '../lib/utils/with_client';
+
+const scenario: Scenario<ApmFields> = async ({ logger, versionOverride }) => {
+  const isLegacy = versionOverride && semver.lt(versionOverride as string, '8.7.0');
+  return {
+    bootstrap: async ({ apmEsClient }) => {
+      if (isLegacy) {
+        apmEsClient.pipeline((base: Readable) => {
+          const defaultPipeline = apmEsClient.getDefaultPipeline()(
+            base
+          ) as unknown as NodeJS.ReadableStream;
+
+          return pipeline(defaultPipeline, deleteSummaryFieldTransform(), (err) => {
```

Comment:
Perhaps also add `observer.version` to indicate which versions of APM Server this affects.

```suggestion
          return pipeline(defaultPipeline, addObserverVersionTransform('8.5.0'), deleteSummaryFieldTransform(), (err) => {
```